ci: test build with external dependencies on every pull request

dundargoc opened the pull request:
- ci: remove environment variable DEPS_BUILD_DIR from CI

If one uses .deps when DEPS_BUILD_DIR is defined in another location it
leads to very surprising behaviors, as it looks for libraries in other
places other than .deps.

- ci: test build with external dependencies on every pull request

Only testing the build with external dependencies on build system
changes is too naive, as demonstrated by
b9f19d3e286d95d9209afbc479fa2eb908067fb1.

- ci: don't install unused packages

The lua client is no longer needed after
d6279f9392073cb1422d76c57baf3fd283ed954e. One of its dependencies,
mpack, is still needed however. Remove lua-nvim and replace it with
lua-mpack.

The other packages are most likely not needed as we no longer run tests
for external dependencies.

justinmk commented on .github/workflows/test.yml:
```diff
@@ -360,3 +359,48 @@ jobs:
+            libuv1-dev \
+            lua-busted \
+            lua-filesystem \
+            lua-inspect \
```
off-topic for this PR, but I wonder if we can eliminate lua-inspect (or is it a transitive dependency of some other luarocks package?)

Probably can eliminate others too, since we use Nvim as the test runner now.

justinmk commented:
> If one uses
> .deps when DPES_BUILD_DIR is defined in another location it leads to
> very surprising behaviors, as it looks for libraries in other places
> other than .deps.

Should we add a comment about that here: https://github.com/neovim/neovim/blob/3ad845882413944a8d6ed3bd9da68c6b4cee0afb/cmake.deps/CMakeLists.txt#L40

justinmk commented on .github/workflows/test.yml:
```diff
@@ -360,3 +359,48 @@ jobs:
+            lua-filesystem \
+            lua-inspect \
+            lua-lpeg \
+            lua-nvim \
```
Is this the lua-client? Is this still needed after https://github.com/neovim/neovim/pull/22077 ?

dundargoc replied:
Yes, it's the lua client. Not needed, but one of its dependencies, mpack, is needed. Removing nvim client and explicitly install mpack instead.

dundargoc committed:
ci: don't install unused packages

The lua client is no longer needed after
d6279f9392073cb1422d76c57baf3fd283ed954e. One of its dependencies,
mpack, is still needed however. Remove lua-nvim and replace it with
lua-mpack.

The other packages are most likely not needed as we no longer run tests
for external dependencies.

dundargoc commented:
> Should we add a comment about that here (or wherever the main declaration of `DEPS_BUILD_DIR` lives):

That is `DEPS_BUILD_DIR` the cmake cache variable, not `DEPS_BUILD_DIR` the environment variable. There is no main declaration place for it as it lives outside of the both cmake project since it's a glue between the two projects.

dundargoc committed:
ci: remove environment variable DEPS_BUILD_DIR from CI

If one uses .deps when DEPS_BUILD_DIR is defined in another location it
leads to very surprising behaviors, as it looks for libraries in other
places other than .deps.

dundargoc committed:
ci: test build with external dependencies on every pull request

Only testing the build with external dependencies on build system
changes is too naive, as demonstrated by
b9f19d3e286d95d9209afbc479fa2eb908067fb1.

dundargoc commented:
> Should we add a comment about that here (or wherever the main declaration of `DEPS_BUILD_DIR` lives):

It's an environment variable meant to glue the two projects together, so it doesn't really have a main declaration. We can add a comment, but I don't think it'll give much insight to say "the DEPS_BUILD_DIR needs to point to the dependency build directory, otherwise it won't be able to find the dependency build directory". This was mostly meant to make CI less surprising, as all jobs share the same environment variables, but not all jobs may build in the same manner.

justinmk commented on .github/workflows/build.yml:
```diff
@@ -71,46 +71,3 @@ jobs:
-            libunibilium-dev \
-            libuv1-dev \
-            lua-busted \
-            lua-filesystem \
```
Technically we still use `require('lfs')` in several places, but I think we accidentally get it from e.g. busted.

I think we could eliminate all those `require('lfs')` eventually, and use `vim.fs` instead.

dundargoc replied:
I see. Should I re-add `lua-filesystem`, or did you mean this as a reference for later?

dundargoc replied:
I've re-added it.
